Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -446,4 +593,182 @@ function handleBitangents(shaderBuilder, attributes) {
+    result = dequantizePositionsTypedArray(
+      attribute.typedArray,
+      attribute.quantization
+    );
+  } else {
+    result = new Float32Array(
+      typedArray.buffer,
+      typedArray.byteOffset,
+      typedArray.byteLength / Float32Array.BYTES_PER_ELEMENT
+    );
+  }
+
+  const startIndex = attribute.byteOffset / Float32Array.BYTES_PER_ELEMENT;
+  const length = result.length;
+  for (let i = startIndex; i < length; i += 3) {
+    const initialPosition = Cartesian3.fromArray(result, i, scratchPosition);
+
+    const projectedPosition = projectPositionTo2D(
+      initialPosition,
+      modelMatrix,
+      projection,
+      initialPosition
+    );
+
+    // To prevent jitter, the positions are defined relative to
+    // a common reference point. For convenience, this is the center
+    // of the primitive's bounding sphere.
+    const relativePosition = Cartesian3.subtract(
+      projectedPosition,
+      referencePoint,
+      projectedPosition
+    );
+
+    result[i] = relativePosition.x;
+    result[i + 1] = relativePosition.y;
+    result[i + 2] = relativePosition.z;
+  }
+
+  return result;
+}
+
+const scratchMatrix = new Matrix4();
+const scratchProjectedMin = new Cartesian3();
+const scratchProjectedMax = new Cartesian3();
+const scratchBoundingSphere = new BoundingSphere();
+
+function modifyAttributeFor2D(attribute, renderResources, frameState) {
+  const sceneGraph = renderResources.model.sceneGraph;
+  const modelMatrix = sceneGraph.computedModelMatrix;
+  const nodeComputedTransform = renderResources.runtimeNode.computedTransform;
+  const computedModelMatrix = Matrix4.multiplyTransformation(
+    modelMatrix,
+    nodeComputedTransform,
+    scratchMatrix
+  );
+  const projection = frameState.mapProjection;
+
+  // Compute the bounding sphere in 2D.
+  const projectedMin = projectPositionTo2D(
+    renderResources.positionMin,
+    computedModelMatrix,
+    projection,
+    scratchProjectedMin
+  );
+  const projectedMax = projectPositionTo2D(
+    renderResources.positionMax,
+    computedModelMatrix,
+    projection,
+    scratchProjectedMax
+  );
+  const boundingSphere = BoundingSphere.fromCornerPoints(
+    projectedMin,
+    projectedMax,
+    scratchBoundingSphere
+  );
+
+  const referencePoint = Cartesian3.clone(
+    boundingSphere.center,
+    new Cartesian3()
+  );
+
+  // Project positions relative to the 2D bounding sphere's center.
+  const projectedPositions = createPositionsTypedArrayFor2D(
+    attribute,
+    computedModelMatrix,
+    projection,
+    referencePoint
+  );
+
+  // Put the resulting data in a GPU buffer.
+  const buffer = Buffer.createVertexBuffer({
+    context: frameState.context,
+    typedArray: projectedPositions,
+    usage: BufferUsage.STATIC_DRAW,
+  });
+  buffer.vertexArrayDestroyable = false;
+
+  // Store the reference point for repeated future use.
+  attribute.buffer2D = buffer;
+  attribute.referencePoint2D = referencePoint;
```

Comment:
Hm, the typical pattern we've been using is to add generated resources to `model._resources`, and whenever the pipeline is rebuilt, those resources are `.destroy()-ed` to start over fresh. Makes it easier to manage the memory, but perhaps is more wasteful if things can be cached. We might want to discuss this more